### PR TITLE
Using glance image instead of openstack image

### PR DESCRIPTION
**Pull request by abel-navarro**
With Liberty, the openstack CLI is not able to retrieve and image by
its name, as it is used in qa_crowbarsetup.sh. This patch substitutes
those calls to glance image commands.

### CODE REVIEW

**Comment by aplanas** [comment]
Can you amend the first commit user?

**Comment by aplanas** [comment]
The second commit is not related, isn't?

**Comment by abel-navarro** [comment]
@aplanas Updated the author and removed space around parenthesis. That second commit was a pull merge instead of a rebase, but I got rid of it.

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2835,6 +2835,18 @@ function adapt_dns_for_docker()
...
 
+function glance_image_exists()
+{
+    glance image-list | grep -q "[[:space:]]$1[[:space:]]"
+    return $? 
+}
+
+function glance_image_get_id()
+{
+    local image_id=$( glance image-list | grep "[[:space:]]$1[[:space:]]" | awk '{ print $2 }' )
```
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: Spaces around parenthesis

**Comment by aplanas** [comment]
@abel-navarro thanks!

+1

**Comment by bmwiedemann** [comment]
@mjura  why did we switch from glange to openstackclient before? in #376

**Comment by dirkmueller** [comment]
-1 glance is deprecated. we can fix it by using the lookup by id, no?

**Comment by abel-navarro** [comment]
@dirkmueller why is glance CLI deprecated? I can still see in the official documentation (Liberty) is the prefered choice:

http://docs.openstack.org/user-guide/common/cli_manage_images.html

**Comment by vmoravec** [comment]
@abel-navarro I'm wondering how much the official docs have change recently. Several months ago I was doing small research among the openstack cli commands for the cct testsuite and most of them (including glance) had a note about to be deprecated in favour of openstack-client all over the docs. These notes seem to be gone now. I just hope the openstack-client is not going to be deprecated instead :)

**Comment by abel-navarro** [comment]
@vmoravec there is at least one service (Keystone) that has [deprecated its CLI] (http://specs.openstack.org/openstack/keystone-specs/specs/keystoneclient/deprecate-cli.html). You cannot see that in other services like Glance. Probably there's a trend to converge all CLI into [OSC](http://docs.openstack.org/developer/python-openstackclient/) but I cannot find any statement regarding this.

Don't get me wrong, I'm not opposing at all to use OSC, just want to understand the motivation.

**Comment by vmoravec** [comment]
@abel-navarro As I said, the docs have changed for Liberty, all those deprecation notes are gone, no idea why. 

>    Probably there's a trend to converge all CLI into OSC but I cannot find any statement regarding this.

That  was actually the official point back then when the deprecation notes where present there. Apparently it's also gone. That means one needs to be pretty careful about the statements in the official OS docs.

**Comment by abel-navarro** [comment]
I was checking documentation. It seems Keystone CLI is on active way of [being](http://docs.openstack.org/developer/python-keystoneclient/man/keystone.html) [deprecated](https://specs.openstack.org/openstack/keystone-specs/specs/keystoneclient/deprecate-cli.html), but there's no active actions from the other groups.

At least OSC is being [actively](https://git.openstack.org/cgit/openstack/python-openstackclient/) maintained so looks safe for me to keep using OSC. Worst case we can revert to the original CLI's, since the syntax is almost the same.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -3042,8 +3048,8 @@ function oncontroller_testsetup()
     fi
 
     # wait for image to finish uploading
-    imageid=`perl -ne "m/ id [ |]*([0-9a-f-]+)/ && print \\$1" glance.out`
-    if [ "x$imageid" == "x" ]; then
+    imageid=$(glance_image_get_id $image_name)
+    if [[ -z $imageid ]]; then
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: This boils down to:
`if ! [[ $imageid ]]`

**Comment by jdsn** [comment]
Even if it does not solve the deprecation of glance, this commit is a nice cleanup and code deduplication, so I propose to merge it anyhow.
The deprecation can be fixed in a later commit also.

So :+1:  from my side

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (abel-navarro) [inline scripts/qa_crowbarsetup.sh]: yep, that's nicer. Updating PR.

**Comment by jdsn** [comment]
So we have a +1 -1 +1 = +1
One +1 is still missing.

**Comment by abel-navarro** [comment]
Well, @dirkmueller comment was about using glance client, but I changed it to openstack client, so his -1 should not apply anymore :)

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -2895,6 +2895,18 @@ function adapt_dns_for_docker()
     neutron subnet-update --dns-nameserver "$dns_server" fixed
 }
 
+function glance_image_exists()
+{
+    openstack image list | grep -q "[[:space:]]$1[[:space:]]"
+    return $?
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: Here is the trailing whitespace that causes the travis validation failure

Comment (abel-navarro) [inline scripts/qa_crowbarsetup.sh]: Thanks for spotting this. Fixing.

**Pull request merged**